pkg/controllers/network_services_controller.go

```diff
@@ -365,14 +363,19 @@ func (nsc *NetworkServicesController) OnEndpointsUpdate(obj interface{}) {
 		return
 	}
 
-	// build new endpoints map to reflect the change
+	glog.V(1).Infof("Received update to endpoints Name: %v in Namespace %v from watch API", ep.Name, ep.Namespace)
+
+	// build new service and endpoints map to reflect the change
+	newServiceMap := nsc.buildServicesInfo()
```

Comment:
this is required, other wise, when kube-router is booting and the `Run()` method of controller is not run yet. In which case we have empty `nsc.serviceMap` and `nsc.endpointsMap` resulting in deletion of endpoints.